src/main/scala/is/hail/variant/ArrayView.scala

```diff
@@ -0,0 +1,25 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+import is.hail.utils._
+
+class ArrayView[V <: View](t: TArray, val elementView: V) {
```

Comment:
I like this.

Just to state the alternative, I thought we could have a `altAlleleRef(i: Int): String` on the variant view that that returned the ref of the `i`th alt allele, etc.  This mirrors the GenotypeView for AD, PL.